### DISCUSSION TITLE
Problems to load several Modelica models using OMPython

### DISCUSSION THREAD

**Discussion by kmerbcn** [OpenModelica/OpenModelica#12444] [General]
Hello

I am  a new user of OMPython. I have an issue trying to load several files using ModelicaSystem().
For example, I have 2 models Model1.mo and Model2.mo . Model1.mo contains Model2.mo . How can I load ModelicaSystem('Model1.mo', 'Model1') and model2 together?

It seems a list is not a solution. Any hints?

Thanks in advance

**Comment by arun3688** [comment]
@kmerbcn You can load dependent files like this 

`>>> ModelicaSystem("model1.mo", "model1", ["model2.mo"])`

see the users guide https://openmodelica.org/doc/OpenModelicaUsersGuide/1.22/ompython.html#enhanced-ompython-features, the third argument is used for loading the library or dependant modelica files

**Reply by kmerbcn** [reply]
Thanks for the help Arun!
More precisely, I have a folder full of *.mo models. In particular using the om shell script I list all the files using a mos scrip and the terminal opens the files using LoadFile instruction.
How can i load, using OMPython, all the files inside of this folder?

Best regards

**Comment by kmerbcn** [comment]
Basically I have the following code

 mod = ModelicaSystem("C:/Users/Javito/Documents/Pruebas_modelica/Electronica_Potencia/BuckConverter.mo", 'BuckConverter',
                         ["C:/Users/Javito/Documents/Pruebas_modelica/Electronica_Potencia/BootStrapDriver.mo",
                          "C:/Users/Javito/Documents/Pruebas_modelica/Electronica_Potencia/MOSFET.mo"
                          "C:/Users/Javito/Documents/Pruebas_modelica/Electronica_Potencia/LevelShift.mo",
                          "C:/Users/Javito/Documents/Pruebas_modelica/Electronica_Potencia/DriverTotemPole.mo",
                          "C:/Users/Javito/Documents/Pruebas_modelica/Electronica_Potencia/NPN.mo",
                          "C:/Users/Javito/Documents/Pruebas_modelica/Electronica_Potencia/PNP.mo"]);

It seems that it does not work.

**Reply by arun3688** [reply]
can you show your error message, I guess based on the above code i can see that your library is a package and if that is true you should basically do
`mod = ModelicaSystem("C:/Users/Javito/Documents/Pruebas_modelica/Electronica_Potencia/package.mo", 'BuckConverter']
`
and it should work

**Comment by kmerbcn** [comment]
Hello Arun

My folder is not a package. Unfortunately, this is an old model I did and before I used to build models connecting other models. The situation is a s follows . The code looks like

 mod = ModelicaSystem("C:/Users/Javito/Documents/Pruebas_modelica/Electronica_Potencia/BuckConverter.mo", 'BuckConverter',
                         ["C:/Users/Javito/Documents/Pruebas_modelica/Electronica_Potencia/BootStrapDriver.mo",
                          "C:/Users/Javito/Documents/Pruebas_modelica/Electronica_Potencia/MOSFET.mo",
                          "C:/Users/Javito/Documents/Pruebas_modelica/Electronica_Potencia/LevelShift.mo",
                          "C:/Users/Javito/Documents/Pruebas_modelica/Electronica_Potencia/FromBooleanToReal.mo",
                          "C:/Users/Javito/Documents/Pruebas_modelica/Electronica_Potencia/DriverTotemPole.mo",
                          "C:/Users/Javito/Documents/Pruebas_modelica/Electronica_Potencia/NPN.mo",
                          "C:/Users/Javito/Documents/Pruebas_modelica/Electronica_Potencia/PNP.mo"
                          "C:/Users/Javito/Documents/Pruebas_modelica/Electronica_Potencia/VariableCap.mo"]);

Error message is :

[C:/Users/Javito/Documents/Pruebas_modelica/Electronica_Potencia/DriverTotemPole.mo:2:3-3:114:writable] Error: Class PNP not found in scope DriverTotemPole.

[C:/Users/Javito/Documents/Pruebas_modelica/Electronica_Potencia/DriverTotemPole.mo:2:3-3:114:writable] Error: Class PNP not found in scope DriverTotemPole.

This is because the hierarchy of Buck converter is as follows : Buck Converter contains BootstrapDrive. The latter contains the blocks Level Shift.mo and DriverTotemPole.mo. The last model contains PNP and NPN.mo . How should I use ModelicaSystem? Should I rewrite all the models in a package?

Best regards,

Javier

**Reply by arun3688** [reply]
@kmerbcn` Class PNP not found in scope DriverTotemPole.` this gives you what is the problem, PNP is not found and may be it is a library,  May be you should first try to load in OMEdit and see if your model works

**Comment by kmerbcn** [comment]
Hello Arun

My model works with all the files in the list and converges to a real situation. The model is 1 year old.
The reason to change to Python is because I want to run parametric study and do further postprocessing with numpy and scipy.
The library PNP is already in the folder and works in OMEdit.

**Reply by arun3688** [reply]
@kmerbcn can you attach your models in a zip file and let me see if i can reproduce the issue

**Reply by kmerbcn** [reply]
Now it works, I put PNP and NPN in the list of ModelicaSystem before the in the DriverTotemPole and it does not show any error.

Best regards

**Comment by kmerbcn** [comment]
[BuckConverter.zip](https://github.com/OpenModelica/OpenModelica/files/15377847/BuckConverter.zip)
 
Hello Aaron

I have attached you the zip files. I hope you find a solution. Thanks!

**Comment by arun3688** [comment]
@kmerbcn Ok great I will close this issue, please reopen it if you face further issue

**Comment by kmerbcn** [comment]
Hello Arun

I have a question concerning how to load the following package

[ControlResonante.zip](https://github.com/user-attachments/files/15568364/ControlResonante.zip)


As soon as I try to load the package with the instruction ModelicaSystem in Ompython i have an error.
Inside of the code I have the models PWM_h and  H_bridge included.

I call the system as follows :

 mod = ModelicaSystem('ModeloInalambrico.mo', 'ModeloInalambrico', 'CargadorInalambrico.mo', 'Transformer.mo')

CargadorInalambrico is a package containing --> PWM_h and H_Bridge models

the error is

ile ~/anaconda3/lib/python3.9/site-packages/OMPython/__init__.py:830 in __init__
    self.getconn = OMCSession()

  File ~/anaconda3/lib/python3.9/site-packages/OMPython/__init__.py:566 in __init__
    self._connect_to_omc(timeout)

  File ~/anaconda3/lib/python3.9/site-packages/OMPython/__init__.py:576 in _connect_to_omc
    from omniORB import CORBA

ModuleNotFoundError: No module named 'omniORB'

How would you instantiate the model?

Thanks in advance!

**Reply by kmerbcn** [reply]
mod = ModelicaSystem('ModeloInalambrico.mo', 'ModeloInalambrico', ['CargadorInalambrico.mo', 'Transformer.mo'])

I tried like this and it seems it worked.

Thanks for this forum discussion. I really enjoy working with talented people!
